java Null Pointer Exception on getVersionedAppName() method thus causing app to crash

<!-- 

Thank you for taking the time to report an ODK Collect issue!

This space is for submitting problems and feature requests. For general usage or form design questions, please email opendatakit@googlegroups.com or for ODK Collect source code questions, please email opendatakit-developers@googlegroups.com

Before filling this form, visit https://github.com/opendatakit/collect/issues?q=is%3Aissue and search to see whether your issue was already reported or fixed. If you find a match, comment on it or add a +1 rather than posting a new issue. If you find a problem you know how to fix, submit a pull request. 🎉

For all problem reports, please use the template below. Also include any relevant stack traces or error messages.

For feature requests, please include the problem description (what problem do you have that can't currently be solved?) and a proposed solution if you have one in mind (optional). You can delete the template. 

-->

#### Software and hardware versions 
Collect v1.x.x, Android v5.1.1, device used...

#### Problem description
App Crashes as soon as it installs.

#### Steps to reproduce the problem

#### Expected behavior

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

I was able to reproduce this issue as well. It takes place when I download the code - if I use `git clone` everything is ok. I think we should catch that null.

> 03-13 11:21:36.520 18178-18178/org.odk.collect.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                         Process: org.odk.collect.android, PID: 18178
                                                                         java.lang.RuntimeException: Unable to start activity ComponentInfo{org.odk.collect.android/org.odk.collect.android.activities.MainMenuActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.String.replaceFirst(java.lang.String, java.lang.String)' on a null object reference
                                                                             at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2702)
                                                                             at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2767)
                                                                             at android.app.ActivityThread.access$900(ActivityThread.java:177)
                                                                             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1449)
                                                                             at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                             at android.os.Looper.loop(Looper.java:145)
                                                                             at android.app.ActivityThread.main(ActivityThread.java:5951)
                                                                             at java.lang.reflect.Method.invoke(Native Method)
                                                                             at java.lang.reflect.Method.invoke(Method.java:372)
                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
                                                                          Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.String.replaceFirst(java.lang.String, java.lang.String)' on a null object reference
                                                                             at org.odk.collect.android.application.Collect.getVersionedAppName(Collect.java:179)
                                                                             at org.odk.collect.android.activities.MainMenuActivity.onCreate(MainMenuActivity.java:130)
                                                                             at android.app.Activity.performCreate(Activity.java:6289)
                                                                             at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
                                                                             at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2655)
                                                                             at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2767) 
                                                                             at android.app.ActivityThread.access$900(ActivityThread.java:177) 
                                                                             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1449) 
                                                                             at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                             at android.os.Looper.loop(Looper.java:145) 
                                                                             at android.app.ActivityThread.main(ActivityThread.java:5951) 
                                                                             at java.lang.reflect.Method.invoke(Native Method) 
                                                                             at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400) 
                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195) 

--- a/collect_app/src/main/java/org/odk/collect/android/application/Collect.java
+++ b/collect_app/src/main/java/org/odk/collect/android/application/Collect.java
@@ -171,7 +171,8 @@ public class Collect extends Application {
     }
 
     public String getVersionedAppName() {
-        String versionName = BuildConfig.VERSION_NAME.replaceFirst("-", "\n");
+        String versionName = BuildConfig.VERSION_NAME;
+        versionName = " " + versionName.replaceFirst("-", "\n");
         return getString(R.string.app_name) + versionName;
     }
 

--- a/collect_app/src/main/java/org/odk/collect/android/application/Collect.java
+++ b/collect_app/src/main/java/org/odk/collect/android/application/Collect.java
@@ -17,7 +17,6 @@ package org.odk.collect.android.application;
 import android.app.Application;
 import android.content.Context;
 import android.content.SharedPreferences;
-import android.content.pm.PackageManager.NameNotFoundException;
 import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 import android.os.Environment;
@@ -27,6 +26,7 @@ import android.support.multidex.MultiDex;
 import com.google.android.gms.analytics.GoogleAnalytics;
 import com.google.android.gms.analytics.Tracker;
 
+import org.odk.collect.android.BuildConfig;
 import org.odk.collect.android.R;
 import org.odk.collect.android.database.ActivityLogger;
 import org.odk.collect.android.external.ExternalDataManager;
@@ -172,15 +172,8 @@ public class Collect extends Application {
 
     public String getVersionedAppName() {
         String versionName = "";
-        try {
-            versionName = getPackageManager()
-                    .getPackageInfo(getPackageName(), 0)
-                    .versionName;
-            versionName = " " + versionName.replaceFirst("-", "\n");
-        } catch (NameNotFoundException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
+        versionName = BuildConfig.VERSION_NAME;
+        versionName = " " + versionName.replaceFirst("-", "\n");
         return getString(R.string.app_name) + versionName;
     }
 

--- a/collect_app/src/main/java/org/odk/collect/android/application/Collect.java
+++ b/collect_app/src/main/java/org/odk/collect/android/application/Collect.java
@@ -171,9 +171,7 @@ public class Collect extends Application {
     }
 
     public String getVersionedAppName() {
-        String versionName = "";
-        versionName = BuildConfig.VERSION_NAME;
-        versionName = " " + versionName.replaceFirst("-", "\n");
+        String versionName = BuildConfig.VERSION_NAME.replaceFirst("-", "\n");
         return getString(R.string.app_name) + versionName;
     }
 

